runtime/strategies/combined-strategy.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+let assert = require('assert');
+let {Strategy} = require('../../strategizer/strategizer.js');
+let Recipe = require('../recipe/recipe.js');
+let RecipeWalker = require('../recipe/walker.js');
+
+module.exports = class CombinedStrategy extends Strategy {
+  constructor(strategies) {
+    super();
+    assert(strategies.length > 1, 'Strategies must contain at least 2 elements.');
+    this._strategies = strategies;
+    this._strategies.forEach(strategy => assert(strategy.walker));
+    assert(this._strategies[0].getResults);
+  }
+  async generate(strategizer) {
+    let results = this._strategies[0].getResults(strategizer);
+    let totalResults = [];
+    this._strategies.forEach(strategy => {
+      totalResults.forEach(r => r.final = true);
```

Comment:
I don't quite understand what's going on with `getResults` and `.final`, can you explain the problem? I'm hoping we can find a way to build `CombinedStrategy` without changes to strategizer.

Reply:
I got rid of .final (strategy-explorer populates missing derivation parents instead)
getResults abstraction is needed for the combined strategy to pass each strategy whatever recipes the it needs to operate on - for most of them is it the strategizer.generated, but search-tokens-to-particles needs strategizer.terminal too